vignettes/web_only/repFilter_v3.Rmd

```diff
@@ -0,0 +1,326 @@
+ - `exact` - looks for exact match in gene names
+ - `by substring`- looks for substring in gene names
+ - `by start` - looks for gene names starting with the some pattern
+ 
+Filter out all clonotypes within samples with V gene 'TRBV1' or 'TRGV11'
+
+```{r example-5.1}
+#method = "by.clonotype", query = list(V.gene = exclude("TRBV1", "TRGV11")), match="exact"
+```
+
+Filter out all clonotypes within samples where V gene name contains substrings 'TRBV1' or 'TRGV11'
+```{r example-5.2}
+#method = "by.clonotype", query = list(V.gene = exclude("TRBV1", "TRGV11")), match="substring"
+
+```
+
+Filter out all clonotypes within samples where V gene name starts with 'TRBV1' or 'TRGV11'
+```{r example-5.3}
+#method = "by.clonotype", query = list(V.gene = exclude("TRBV1", "TRGV11")), match="startswith"
+```
+
+## Using `repFilter` function for repertoire data analysis 
+
+We will use `scdata` datasets from **Immunarch** library. Load them by executing the following command:
+```{r}
+data(scdata)
+```
+
+
+Look at meta data from `scdata` datasets and create a new dataset with cluster-specific immune repertoires (for more information see *Single-cell* tutorials):
+
+```{r}
+scdata_cl <- select_clusters(scdata, scdata$bc_cluster, "Cluster")
+scdata_cl$meta
+```
+
+And compare the number of clonotypes between clusters:
+
+```{r clonotypes, warnings = F}
+vis(repExplore(scdata_cl$data, .method = "volume"))
+```
+
+Compare J gene usage between 3 cluster:
+
+```{r geneUsage,  warning=F}
+# method = "by.meta", query = list(Cluster = include('Activ'))
+sc_active <- geneUsage(scdata_cl[1], "hs.trbj", .norm = T)
+p1 <- vis(sc_active)
+p1
+# method = "by.meta", query = list(Cluster = include('Memory'))
+sc_memory <- geneUsage(scdata_cl[1], "hs.trbj", .norm = T)
+p2 <- vis(sc_memory)
+p2
+# method = "by.meta", query = list(Cluster = include('Naive'))
+sc_naive <- geneUsage(scdata_cl[1], "hs.trbj", .norm = T)
+p3 <- vis(sc_naive)
+p3
+
+```
+
+Compare gene usage of IGHV3 between 3 cluster:
+```{r IGHV3, warning=F}
+# method = "by.meta", query = list(Cluster = include('Activ'))
+# method = "by.gene", query = list(J = include('IGHJ4'))
+sc_active <- geneUsage(scdata_cl$data$flu_Activ %>% filter(grepl('IGHV3',V.name)), "hs.trbj", .norm = T)
+p1 <- vis(sc_active)
+# method = "by.meta", query = list(Cluster = include('Memory'))
+# method = "by.gene", query = list(J = include('IGHJ4'))
+sc_memory <- geneUsage(scdata_cl$data$flu_Memory %>% filter(grepl('IGHV3',V.name)), "hs.trbj", .norm = T)
+p2 <- vis(sc_memory)
+# method = "by.meta", query = list(Cluster = include('Naive'))
+# method = "by.gene", query = list(J = include('IGHJ4'))
+sc_naive <- geneUsage(scdata_cl$data$flu_Naive %>% filter(grepl('IGHV3',V.name)), "hs.trbj", .norm = T)
+p3 <- vis(sc_naive)
+p1
+p2
+p3
+
+```
+
+Look at the coding and nonconding clonotypes in each cluster. As can be easily seen, there is no noncoding clonotypes in the datasets:
```

Comment:
As you can see, there aren't any noncoding clonotypes in the datasets // Note that there aren't any noncoding clonotypes...